Sources/HaCWebsiteLib/Controllers/EventApiController.swift

```diff
@@ -0,0 +1,97 @@
+import Kitura
+import KituraNet
+import SwiftyJSON
+import Foundation
+import DotEnv
+
+struct EventApiController {
+  static var handler: RouterHandler = { request, response, next in
+    response.headers["Content-Type"] = "text/plain; charset=utf-8"
+    guard let parsedBody = request.body else {
+      next()
+      response.statusCode = HTTPStatusCode.internalServerError
+      try response.send("Sorry - we weren't able to parse the body of the request\n").end()
+      return
+    }
+    if case .json(let json) = parsedBody {
+      do {
+        try saveEvent(json: json)
+        try response.send("Successfully added your event to the database\n").end()
+      }
+      catch EventParsingError.missingParameters {
+        response.statusCode = HTTPStatusCode.badRequest
+        try response.send("Sorry - looks like you didn't include all the necessary fields\n").end()
+      } 
+    } else {
+      response.statusCode = HTTPStatusCode.badRequest
+      try response.send("Please use JSON for post data\n").end()
+    }
+    next()
+  }
+
+  static func saveEvent(json: [String : Any]) throws {
+    let event = try parseEvent(json: json)
+    try event.save()
+  }
+
+  static func parseEvent(json : [String : Any]) throws -> GeneralEvent {
+    let dateFormatter = DateFormatter()
+    dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.zzz"
+
+    guard let title = json["title"] as? String,
+    let startDateString = json["startDate"] as? String,
```

Comment:
@CharlieCrisp I know `Encodeable` is giving you grief atm but would be worthwhile exploring later. Could you open an issue to further explore doing this all as encodeable?

Reply:
Issue has been opened :)